app/models/course/assessment/category.rb

```diff
@@ -37,11 +37,11 @@ def other_categories_remaining?
 
   def build_initial_tab
     tabs.build(title: Course::Assessment::Tab.human_attribute_name('title.default'),
-               weight: 0, category: self)
+               weight: 0, category: self) if tabs.empty?
   end
 
   def build_initial_folder
-    build_folder(start_at: Time.zone.now)
+    build_folder(start_at: Time.zone.now) unless folder
```

Comment:
I think we want to be precise. If there's a specific condition we are checking for, that should be checked.